core/src/main/java/io/ddf/etl/ATimeSeriesHandler.java

```diff
@@ -0,0 +1,127 @@
+package io.ddf.etl;
+
+
+import io.ddf.DDF;
+import io.ddf.analytics.ABinningHandler.BinningType;
+import io.ddf.exception.DDFException;
+import io.ddf.misc.ADDFFunctionalGroupHandler;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import com.google.common.collect.Lists;
+
+public abstract class ATimeSeriesHandler extends ADDFFunctionalGroupHandler implements IHandleTimeSeries {
+
+  protected String mTimestampColumn;
+  protected String mTsIDColumn = null;
+
+
+  public ATimeSeriesHandler(DDF theDDF) {
+    super(theDDF);
+
+  }
+
+  public void setTimeStampColumn(String colName) {
+    mTimestampColumn = colName;
+  }
+
+  public String getTimeStampColumn() {
+    return mTimestampColumn;
+  }
+
+
+  public String getTsIDColumn() {
+    return mTsIDColumn;
+  }
+
+  public void setTsIDColumn(String colName) {
+    this.mTsIDColumn = colName;
+  }
+
+  @Override
+  public DDF downsample(String timestampColumn, List<String> aggregateFunctions, int interval, TimeUnit timeUnit)
+      throws DDFException {
+
+    this.mTimestampColumn = timestampColumn;
+    List<String> groupByCols = Lists.newArrayList(timestampColumn);
+    if (mTsIDColumn != null && !mTsIDColumn.isEmpty()) {
+      groupByCols.add(mTsIDColumn);
+    }
+
+    long intervalInSeconds = timeUnit.toSeconds(interval);
+
+    int numBins = getNumBins(intervalInSeconds);
+    DDF binnedDDF = this.getDDF().binning(timestampColumn, BinningType.EQUALINTERVAL.toString(), numBins, null, false,
+        true, true);
+    DDF newDDF = binnedDDF.groupBy(groupByCols, aggregateFunctions);
+
+    return newDDF;
+  }
+
+  @Override
+  public DDF downsample(String timestampColumn, String tsIDColumn, List<String> aggregateFunctions, int interval,
+      TimeUnit timeUnit) throws DDFException {
+
+    this.mTsIDColumn = tsIDColumn;
+    List<String> rs = getDistinctValues(tsIDColumn);
+
+    DDF ddf0 = filterByValue(tsIDColumn, rs.get(0));
+
+    ddf0.getTimeSeriesHandler().setTsIDColumn(tsIDColumn);
+    DDF newDDF = ddf0.getTimeSeriesHandler().downsample(timestampColumn, aggregateFunctions, interval, timeUnit);
+    if (rs.size() > 1) {
+      for (int i = 1; i < rs.size(); i++) {
```

Comment:
There 1 key for most usecases for now. I wanted to quickly reuse DDF binning method here. Other KPIs like col diff or moving average used SparkDF window functions.